.github/workflows/ci.yml

```diff
@@ -0,0 +1,83 @@
+name: CI
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    name: Build
+    strategy:
+      matrix:
+        go-version: [1.15.x, 1.14.x]
+        platform: [ubuntu-latest, macos-latest]
+        env:
+          - TAGS=""
+          - TAGS="-tags bounds"
+          - TAGS="-tags noasm"
+          - TAGS="-tags safe"
+          - FORCE_GOARCH=386
+        exclude:
+          - platform: macos-latest
+            env: TAGS="-tags bounds"
+          - platform: macos-latest
+            env: TAGS="-tags noasm"
+          - platform: macos-latest
+            env: TAGS="-tags safe"
+          - platform: macos-latest
+            env: FORCE_GOARCH=386
+
+    runs-on: ${{ matrix.platform }}
+    env:
+        GO111MODULE: on
+        GOPATH: ${{ github.workspace }}
+    defaults:
+        run:
+            working-directory: ${{ env.GOPATH }}/src/gonum.org/v1/gonum
+
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Cache-Go
+      uses: actions/cache@v1
+      with:
+        path: |
+            ~/go/pkg/mod              # Module download cache
+            ~/.cache/go-build         # Build cache (Linux)
+            ~/Library/Caches/go-build # Build cache (Mac)
+            '%LocalAppData%\go-build' # Build cache (Windows)
+
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+          path: ${{ env.GOPATH }}/src/gonum.org/v1/gonum
+
+    - name: Check copyrights+imports+formatting+generate
+      if: matrix.platform == 'ubuntu-latest' && matrix.tags == ''
```

Comment:
I guess this relies on the assumption that the test variants are executed in the same order than the `matrix/env` list.
I guess that's a rather solid assumption.

Reply:
It would be a very surprising behaviour if they didn't, it maps to an array. We'll know if this doesn't work at any point and we can fix it them.